File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, j, cnt = 0, len(nums) - 1, 0
        while i <= j:
            if nums[j] == val:
                j -= 1
                cnt += 1
                continue
            elif nums[i] == val:
                nums[i], nums[j] = nums[j], nums[i]
                cnt += 1
                j -= 1
            i += 1
        
        return len(nums) - cnt

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int i = 0, j = nums.size() - 1, cnt = 0;
        
        while (i <= j){
            if (nums[j] == val){
                --j;
                ++cnt;
                continue;
            }
            else if (nums[i] == val){
                swap(nums[i], nums[j]);
                ++cnt;
                --j;
            }
            ++i;
        }
        
        return nums.size() - cnt;
    }
};